[RNMobile][Monorepo] Update react-native-* packages from gutenberg-mobile

Tug opened the pull request:
## Description
This PR's goal is to bring the latest changes from gutenberg-mobile to the monorepo branch.
Thus, it also gets gutenberg `master` recent changes in are the 2 repo needs to be in sync.

Most commits are from the gutenberg-mobile repo, the ones that are not are merge commits (from gb-mobile develop or gb master)

> Note: We have a large amount of changes in gutenberg-mobile due to the recent react-native upgrade.

## Testing Instructions
Try running the following commands:
- `npm run native test`
- `npm run test-unit:native`
- `npm run native start:reset` then `curl --silent --output /dev/null "http://localhost:8081/index.bundle?platform=android&dev=true&minify=false"` to generate a bundle for android
- Check CI jobs on this PR, only the "License compatibility" one should be failing (this is still to be addressed)

## Types of changes
Port of commits

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [ ] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [ ] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [ ] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR. <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/native-mobile.md -->

dratwas commented on packages/react-native-editor/metro.config.js:
```diff
@@ -0,0 +1,24 @@
+	},
+};
+
+console.log( 'metro config ', module.exports );
```
Do you want to leave it here or is it temporary?

Tug replied:
Good point, this should be removed

dratwas commented:
>npm run start:reset

I believe that this should be `npm run native start:reset`
With that command, everything seems to work.

Also, I'm not sure if we should leave the console log with the whole metro config at the start. Overall it looks good :)

gziolo commented:
@Tug and @dratwas – do you want me to validate anything specific?

koke committed:
Added entry to release notes

mkaz committed:
Docs: Fixes links in platform and custom block editor tutorial (#20686)

* Fix links for platform docs

* Fix link

* One more...

aduth committed:
Revert "Add Patterns:"

This reverts commit 756a7222332304372a98205f492fa669cd4f5bf3.

cliffordp committed:
Docs: Correct link to list of WP's supported browsers (#20712)

I assume this is where it should be pointing.

koke committed:
Update gutenberg

SergioEstevao committed:
Update gutenberg reference.

gziolo committed:
Block library: Use block.json consistently for FSE blocks (#20717)

koke committed:
Update bundles

aduth committed:
Block Library: Flatten width calculation of gallery columns (#20690)

* Block Library: Flatten width calculation of gallery columns

* Block Library: Gallery: Split gutter by number of columns

youknowriad committed:
Adapt the height of the block preview container to its content (#20715)

mtias committed:
Add basic patterns (#20724)

* Add Patterns:
- Text in two columns.
- Cover block with large text and gradient.

* Fix the phpcs tests

Co-authored-by: Riad Benguella <benguella@gmail.com>

jasmussen committed:
Classic document outline (#20720)

getdave committed:
Update custom block editor tutorials link to point at new platform section (#20722)

Commit:
GradientPicker: Fix control point UI and drag interaction (#20670)

mtias committed:
Add a pattern showing two buttons with styles. (#20726)

jasmussen committed:
Inserter padding, add more left and right. (#20730)

Alternative to #20685.

This one adds padding left and right. Props @shaunandrews.

draganescu committed:
[LatestPosts] Don't trim manual excerpts (#20432)

* adds raw excerpt to the LatestPosts block and respects manual excerpts

* removed the reset to raw excerpt

youknowriad committed:
Bump plugin version to 7.7.0-rc.1

WunderBart committed:
Docs/SlotFills: Small corrections + prettify code examples (#20727)

* Docs/SlotFills: Small correction + prettify code examples

* Tweak individual SlotFill readmes

youknowriad committed:
Add two images side by side pattern (#20725)

aduth committed:
Build: Include entire `lib/` folder in plugin bundle (#20736)

Commit:
RangeControl: Fix Reset and initialPosition (#20247)

* RangeControl: Fix reset and initialPosition logic

* Add tests for desired initialPosition, value, and reset scenarios

* Update storybook snapshots

* Fix null value fill for marks

* Remove unnecessary ref dependency in useEffect hook

* Adjust Range x Tooltip interactions

So that hovering on a tooltip no longer keeps it open

mkaz committed:
Docs: Add Custom Block Editor to TOC and Manifest (#20749)

mkaz committed:
Docs: Add tutorial file to TOC (#20750)

jorgefilipecosta committed:
chore(release): publish

 - @wordpress/block-directory@1.5.4
 - @wordpress/block-editor@3.7.4
 - @wordpress/block-library@2.14.4
 - @wordpress/e2e-tests@1.12.6
 - @wordpress/edit-post@3.13.5
 - @wordpress/edit-site@1.3.3
 - @wordpress/edit-widgets@0.12.4
 - @wordpress/editor@9.12.4
 - @wordpress/format-library@1.14.4

pereirinha committed:
Expand create block options and add readme.txt template (#20694)

* Add Author, License and Version prompt options

* Add readme.txt template

* Fix typo

* Add License URI option

* Do not filter, just assume the default value

* Update default values

* Use version option in documentation

* Update the default description field to add details on limits

* Update readme description

* Remove extra spaces

* Update packages/create-block/lib/prompts.js

* Apply suggestions from code review

Co-authored-by: Grzegorz (Greg) Ziółkowski <grzegorz@gziolo.pl>

ellatrix committed:
Lighter block DOM: Group (#20586)

ellatrix committed:
Lighter block DOM: Navigation (#20729)

* Lighter block DOM: Navigation

* Polish

* Polish

* Fix classes

* Allow to define appender tag name

* Revert link tag for now

mchowning committed:
Revert back to using extracted editing UI in landscape mode on Android (#1989)

* Revert "Merge pull request #1842 from wordpress-mobile/issue/1829_soft-keyboard-landscape"

This reverts commit 5041096d663dda5e014a723e9482f0cc9213eb02, reversing
changes made to ddb8b3fd1553dd57e528dcc0029b2bc014f585ed.

* Remove release note for reverted landscape mode change

etoledom committed:
Update rn-keyboard-aware-scroll-view to point to master HEAD

etoledom committed:
Update gutenberg ref

jasmussen committed:
Try: Show inserter only when block selected for nesting contexts. (#20753)

* Try: Show inserter only when block selected for nesting contexts.

Starts work on #20728. It doesn't completely fix it though.

What it does is create a generic rule to hide the appender in nesting contexts, until the parent or a child of the same block, is selected.

* Fix rebase issue, account for "lighter dom" blocks also.

jasmussen committed:
Polish poster image button arrangement. (#20754)

Fixes #20704.

This changes the phrasing, the button types, and the arrangement of the video poster buttons.

WunderBart committed:
Docs/SlotFills: Small update for consistency (#20767)

etoledom committed:
Merge remote-tracking branch 'origin/develop' into issue/fix-caret-scroll-calculation

SergioEstevao committed:
Update GB reference.

mapk committed:
Fixes #11079. Removed width restriction in CSS for textarea. (#20624)

etoledom committed:
Update gutenberg ref

noahtallen committed:
Add codeowners for env package (#20667)

Adds @noahtallen, @noisysocks, and @epiqueras as codeowners of the wp-env package

SergioEstevao committed:
Update project to use react-native-url-polyfill package for the URL class

SergioEstevao committed:
Update node version on circle ci.

jasmussen committed:
Update Navigation Menu Item icon (#20763)

* Update navigation link to use map marker.

Also update the map marker icon.

* Add triangle margin indicator to menu items.

* Update snapshots.

SergioEstevao committed:
Implement mock block-wrapper for mobile version. (#20772)

SergioEstevao committed:
Update GB reference.

SergioEstevao committed:
Merge pull request #1994 from wordpress-mobile/issue/fix_master_breakage

Fix Web related breakage for using block wrappers

etoledom committed:
Check for `node.classList` to avoid crash on mobile pasting wrapped images. (#20499)

maxme committed:
Set flex-shrink to 0 on the cell container (#20768)

jasmussen committed:
Polish a few broken things in the sidebar. (#20782)

There were some issues with the block sidebar:

- The active style looked a bit overwhelming.
- The button-group control lacked a little style.
- The Width/Height controls for an image were offset.

youknowriad committed:
Bump plugin version to 7.7.0

etoledom committed:
Update gutenberg ref

etoledom committed:
Update gutenberg ref

Commit:
Update Gutenberg ref

chipsnyder committed:
Merge remote-tracking branch 'origin/develop' into issue/1746-latest-post

Commit:
Gallery: Update UI of controls (#20776)

* Reduce Gallery custom control border radius

* Update Gallery control icons to use Chevron (left/right) from @wordpress/icons

chipsnyder committed:
Update gutenberg

Ringish committed:
Adds filtering by multiple categories on LatestPosts block (#20781)

* Implements FormTokenField to LatestPosts block inspector controls, replacing SelectControl

* Implements the posts query with multiple categories

* Improves code & resolves @draganescu's feedback

* Solves PHP Linting

jasmussen committed:
Fix auto-hiding appender regression. (#20780)

In #20753 I caused a regression that collapsed the big square appender that exists in an empty columns block, or an empty group block.

The only thing that should collapse, until selected, is the black appender button inside blocks like Social Links, Buttons, or Navigation Menu.

This PR fixes that regression by scoping the animation to said button.

enriquesanchez committed:
Reduce gap between block library and preview (#20777)

* Reduce gap between block library and preview

* Add border radius and box shadow to block preview popover

* Added variables instead of hard-coded values

chipsnyder committed:
[RNMobile] Latest Posts Block v1 Support in Mobile (#20301)

* Display basic support block for Latest Posts

* Update UI of Latest Posts Placeholder

* Add Post Content Control to the Latest post config menu

* Add support for Post Content Settings and Post Meta Data settings

* Expose QueryControls to Mobile

* Add Post Sorting and options for Latest Post Block

* Refactor LatestPost to allow hooks in component mounting for api requests

* Fix Style issue

* Add Network call on Latest-Posts to fetch categories

* Reorder Options in Latest Posts

* Show configure menu in Latests Posts on focus and on tap of configure

* Reorganize consts

* Reorganize consts

* Mmove web refrences to use defined constants

* Reorder Latest Post Settings to match Web

* Adjust setting for Categories selection

* Fix Prettier issue

* Resolve PHP style issue

* Adjust data parsing for Android

* Adjust the way of identifying the way to parse categoriesList

* Remove Clear Settings and rename the configure button to customize

* Remove function that automatically prompts the settings when tapping on the block

* Implement text and control changes from design feedback

* Adjust separators for Latest Post

* Adjust latest-post post content label to be more excerpt centric, and update wording of related control.

* Update label of excerpt length on latest-posts

* Adjust casing of customize label

* Align customize style with add media empty state

* Fix style issue

* Move set attribute events out of the render function

* Add developer comment for platform differences

* Refactor syntax on set functions

* Modify inspector controls to follow the style of the attribute setters

* Update design of block title

* Update spacing around block

* Add type checking to block non array responses from categoreis request

chipsnyder committed:
Update gutenberg reference

aduth committed:
URL: Use test data from web-platform-tests for isURL spec conformance (#20537)

* URL: Use test data from web-platform-tests for isURL spec conformance

* URL: Change fetch utility to parse JSON

Early iterations included LICENSE fetching as string, but this was later changed to static file. As such, the utility is better served to include JSON parsing.

* URL: Stop fetch execution when rejecting by status code

Interestingly, rejecting (throwing) does not itself halt execution.

* URL: Update fetchJSON function description

* URL: Try using react-native-url-polyfill

* URL: Clarify default exceptions as about:blank parameter specific

* URL: Add Native-specific isURL exceptions

* URL: Move WPT license into explanatory README.md file

Commit:
Merge pull request #1931 from wordpress-mobile/issue/1746-latest-post

Latest Posts Block v1 Support in Mobile

gziolo committed:
Create block: Improve how prompts and values provided are handled (#20756)

SergioEstevao committed:
Update GB reference.

SergioEstevao committed:
Update GB reference.

sainthkh committed:
Trim input value in navigation search input field (#19832)

* Trimmed inputValue.

* Added unit test.

* Trim text in TextHighlight

cameronvoell committed:
Merge branch 'develop' into issue/update_isURL_to_use_polyfill

cameronvoell committed:
Merge pull request #1996 from wordpress-mobile/issue/update_isURL_to_use_polyfill

Update project to use react-native-url-polyfill package for the URL class

dratwas committed:
[RNMobile] Cover block -  Support adding video background from media library (#20463)

* support video background that is set on the web

etoledom committed:
Merge remote-tracking branch 'origin/develop' into issue/fix-crash-pasting-wrapped-images

etoledom committed:
Update RELEASE-NOTES.txt

youknowriad committed:
Update the inserter's block preview to use the AutoHeightPreview (#20817)

draganescu committed:
Adds always on display of media URL (#19504)

* rebased and addressed feedback review

* removed useless space in button label

* use a correct type for empty settings

Co-Authored-By: Andrew Duthie <andrew@andrewduthie.com>

* fixes arrow keys navigation

* adds sending null when suggestions are not needed as it is proper in react rendering

* fixes the ESCAPE key to close the dropdown menu

* Update packages/block-editor/src/components/media-replace-flow/style.scss

Co-Authored-By: Sören Wrede <soerenwrede@gmail.com>

* Update packages/block-editor/src/components/media-replace-flow/style.scss

Co-Authored-By: Sören Wrede <soerenwrede@gmail.com>

* Update packages/block-editor/src/components/media-replace-flow/style.scss

Co-Authored-By: Sören Wrede <soerenwrede@gmail.com>

* Update packages/block-editor/src/components/media-replace-flow/style.scss

Co-Authored-By: Sören Wrede <soerenwrede@gmail.com>

* removes deprecated mixing

* moves keyboard capture outside of LinkControlto account only for it's usage in a dropwdown for MediaFlow

* Overriding the width, removing the hr, and cleaning up spacing with the new grid variables.

* Adjusting the spacing around the input for editing the link URL.

* I'm not sure these are actually doing anything.

Co-authored-by: Andrew Duthie <andrew@andrewduthie.com>
Co-authored-by: Sören Wrede <soerenwrede@gmail.com>
Co-authored-by: Shaun Andrews <shaun@automattic.com>

dratwas committed:
Support video background in the cover block (#1959)

youknowriad committed:
Fix drag and drop into the Columns block (#20822)

youknowriad committed:
Fix columns block selection (#20823)

jasmussen committed:
Polish date-picker component (#20824)

* Refine the buttongroup.

* Refactor am/pm picker to be buttongroup.

* Polish padding.

Fixes #20774.

* Try alternate fix.

* Fix to pass test, and update snaps.

etoledom committed:
Merge pull request #1964 from wordpress-mobile/issue/fix-crash-pasting-wrapped-images

Fix crash pasting HTML with wrapped images.

youknowriad committed:
Fix applyting backgrounds and colors to the columns block in the editor (#20818)

aduth committed:
E2E Tests: Mock Embed response for InnerBlocks locking test (#20481)

aduth committed:
E2E Test Utils: Improve durability of embedding matcher (#20811)

* E2E Test Utils: Improve durability of embedding matcher

* E2E Test Utils: Add newline between functions

* E2E Test Utils: Avoid normalizing path endings

(a) It's not consistent with preceding `parameterEquals` testing and (b) there's not variance here, for all requests via `getEmbedPreview` implementation

See: https://github.com/WordPress/gutenberg/blob/3848806ccc7b8a65778366e80af7ea63258538d4/packages/core-data/src/resolvers.js#L112

aduth committed:
URL: Fix getQueryString incorrect handling of hash fragment (#20738)

* URL: Fix getQueryString incorrect handling of hash fragment

* URL: Update getQueryString test cases to proper URLs

* URL: Add test case for empty querystring as undefined

Previously _sort of_ existed, but as an invalid URL which would fail to construct as `new URL` in the current implementation. Now split to two assertions verifying "invalid URL" behavior and "empty querystring" behavior.

aduth committed:
ESLint Plugin: Relax `prefer-const` for destructuring assignment (#20737)

youknowriad committed:
Bump plugin version to 7.7.1

SergioEstevao committed:
Refactor spacer block to share edit method with the web. (#20746)

* Refactor spacer block to share edit method with the web.

* Update code to now allow spacer larger than 500px.

* Fix constant naming format.

SergioEstevao committed:
Update GB reference.

SergioEstevao committed:
Merge branch 'develop' into issue/refactor_spacer_block

sgomes committed:
Improve performance testing (#20802)

SergioEstevao committed:
Merge pull request #1991 from wordpress-mobile/issue/refactor_spacer_block

Refactor the spacer block.

dratwas committed:
[RNMobile] Fix replacing empty paragraph in inner block (#20653)

* Check if the replaceable block is a nested one

shiki committed:
Update gutenberg ref to f0f6ce3

Commit:
Improves RTL style conversion (#20503)

* Improves rtl style conversion + adds tests

This update improves the rtl style utility to better handle varying combinations of `left` and `Left` based properties. A handful of tests have been added to ensure style property conversions are outputted as expected.

* Adds support for left->right and right->left.

Also adds tests

* Components: Add RTL conversion edge case test

* Rename rtl.test.js to rtl.js

* Fixes upright CSS property use case. Adds tests for scroll padding/margin

* Converts Regex to const. Refactors left/right key conversion utility.

* Adds and improves comments for rtl functions.

* Simplify with lodash.mapKeys and rename function to match standards

Co-authored-by: Andrew Duthie <andrew@andrewduthie.com>

dratwas committed:
Fix replacing empty paragraph in inner block (#1984)

Commit:
Update Gutenberg ref

ceyhun committed:
Point gutenberg submodule to fork

ceyhun committed:
Update gutenberg ref

ceyhun committed:
Point gutenberg submodule to fork

aduth committed:
Revert "Framework: Travis: Avoid skipping Puppeteer download (#20547)" (#20828)

This reverts commit 371faf7ac32a65f5f9ab1cf9bf7a1f9fede47d06.

aduth committed:
Framework: Use WHATWG URL in place of legacy url module (#19823)

* Block Editor: Use WHATWG URL in place of legacy url module

* Block Library: Use WHATWG URL in place of legacy url module

* E2E Test Utils: Block Editor: Use WHATWG URL in place of legacy url module

* E2E Tests: Use WHATWG URL in place of legacy url module

aduth committed:
Compat: Use core-js-url-browser for URL polyfill (#20225)

* Compat: Fix URL polyfill test

* Compat: Use core-js-url-browser for URL polyfill

* Compat: Respect SCRIPT_DEBUG for minified suffix

* Compat: Move polyfill script registration to client-assets.php

Only scripts registered in gutenberg_register_vendor_scripts are downloaded as part of the plugin build script.

etoledom committed:
Update rn-keybaord-aware-scroll-view reference for testing

talldan committed:
Fix visually hidden classnames (#20649)

* Add tests with failing test case

* Allow visually hidden class names to be extended

* Repace manual tests with storybook snapshots

* Fix dodgy grammar

gziolo committed:
Dependencies webpack plugin: Let the output file be specified when output is combined (#20844)

grappler committed:
Add ESNext example for unregisterBlockType (#20784)

Add an explanation why 'wp-edit-post' is needed to unregister a block

gziolo committed:
Docs: Add the release date to the @wordpress/scripts CHANGELOG

Commit:
Update Gutenberg ref

Commit:
Update Gutenberg ref

draganescu committed:
Show errors in the media replace control (#19244)

* adds errors to the media replace control

* adds a debounced speak message

* removes __unstableHTML dependency, uses built in speak, creates notice with a small delay, VO works

* adds comments to setTimeout call for notice and HTML removal on notice error, adds prefix to notice uniqueid

* updates comment wording

* fixes rebased style for media replace flow

koke committed:
Update Gutenberg

ceyhun committed:
Update gutenberg ref

koke committed:
Update Gutenberg

koke committed:
Update Gutenberg

koke committed:
Merge branch 'develop' into enable-page-templates

ceyhun committed:
[RNMobile]: Refine block settings' separator margin (#20842)

* Hide last separator of video block settings

* Remove block settings' cell's left margin when there's no icon

ceyhun committed:
Update 'Add Block Here' indicator style (#20864)

koke committed:
Remove feature flag for mobile page templates (#20718)

* Enable page templates on production builds

* Remove mobileEnablePageTemplates from docs

* Fix contact emoji

* Update spacer height on services template

* Remove old documentation for the feature flag

* Remove unused variable

koke committed:
Update Gutenberg

ceyhun committed:
Revert "Point gutenberg submodule to fork"

This reverts commit a724f6387f475bf40d161e16395d87925cb69942.

koke committed:
Update Gutenberg

etoledom committed:
Update rn-keyboard-aware-scroll-view to version gb-v0.8.8

koke committed:
Merge pull request #1987 from wordpress-mobile/enable-page-templates

Enable page templates

etoledom committed:
Update gutenberg ref

etoledom committed:
Merge remote-tracking branch 'origin/develop' into issue/fix-caret-scroll-calculation

ceyhun committed:
Update gutenberg ref

ceyhun committed:
Merge branch 'develop' into issue/1796-refine-video-settings-separators

ceyhun committed:
Revert "Point gutenberg submodule to fork"

This reverts commit 8852b8957e4b9a241c1c04b5b68f39199696334e.

ceyhun committed:
Update gutenberg ref

lukewalczak committed:
[RNMobile] Create cross platform `useResizeObserver()` hook (#19918)

* Create a hook to be useResizeObserver() and unify with web API

mcsf committed:
Edit Post: Register block patterns as separate plugin (#20871)

Commit:
RNMobile - Replace Icon in toolbar for media editing (#20841)

* RNMobile - Replace Icon - Removes duplicated icon and adds it to the icons library, used in Cover, Media & Text and Image block

* Storybook - Update snapshot

* RNMobile - Video block - Update edit media icon

* Icons - Update Replace Icon

* Storybook - Update snapshot

Commit:
Merge branch 'develop' of github.com:wordpress-mobile/gutenberg-mobile into feature/replace-icon

# Conflicts:
#	gutenberg

Commit:
Update Gutenberg ref

Commit:
Merge pull request #2007 from wordpress-mobile/feature/replace-icon

Replace Media Icon

ntwb committed:
Add Prettier shared config package (#20026)

* Add Prettier shared config package

* Update packages/prettier-config/README.md

Co-Authored-By: Grzegorz (Greg) Ziółkowski <grzegorz@gziolo.pl>

* Update README.md

* Use @wordpres/prettier-config with @wordpress/scripts

Co-authored-by: Grzegorz (Greg) Ziółkowski <grzegorz@gziolo.pl>

ceyhun committed:
Merge branch 'develop' into issue/1796-refine-video-settings-separators

ceyhun committed:
Merge pull request #2008 from wordpress-mobile/issue/1796-refine-video-settings-separators

Refine block settings' separator margin

ceyhun committed:
Merge branch 'develop' into issue/1166-update-add-new-block-indicator-style

ceyhun committed:
Update gutenberg ref

obenland committed:
Patterns: Make adding patterns easier (#20854)

ceyhun committed:
Merge branch 'develop' into try/speed-up-uitests

maxme committed:
Update cellRowStyles to add flex-shrink back and fix a disapearing stepper in bottom sheets (#20878)

youknowriad committed:
Add default styles to the TabPanel component (#20872)

Tug committed:
[RNMobile] Remove left border of the toolbar (#20830)

etoledom committed:
Update gutenberg ref

etoledom committed:
Merge remote-tracking branch 'origin/develop' into issue/fix-caret-scroll-calculation

shaunandrews committed:
Preview Button: Remove the separator and border, and reduce the size of the icon. (#20683)

* Removes the separator and border from the preview toggle.
* Updates button to always says "Preview".
* Adjust external link icon.
* Removed the "View" group label and put the external icon first.

etoledom committed:
[RNMobile] Adding global toolbar height to auto-scroll calculations (#20334)

* Adding global toolbar height to auto-scroll calculations

* Giving a name to magic number to reflect the intention

* Using inputAccessoryViewHeight on KeyboardAwareFlatList for iOS scrolling calculations.

* Fix syntax styling

* Avoid using local constants for component dimensions

* Fix lint issues

* Using new dimension constants everywhere.

* Add new global styles to mocks

etoledom committed:
Update gutenberg ref

ceyhun committed:
Merge pull request #2014 from wordpress-mobile/issue/1166-update-add-new-block-indicator-style

Update 'Add Block Here' indicator style

etoledom committed:
Merge remote-tracking branch 'origin/develop' into issue/fix-caret-scroll-calculation

ceyhun committed:
Merge branch 'develop' into try/speed-up-uitests

etoledom committed:
Merge pull request #1939 from wordpress-mobile/issue/fix-caret-scroll-calculation

[iOS] Fix autoscroll behaviour.

vindl committed:
Site Editor: style resets for top level page (#20886)

Allow Site Editor to be loaded as top level page.

jorgefilipecosta committed:
Fix: ResponsiveWrapper can not handle cases with small dimensions

ianstewart committed:
Update text-two-columns.json (#20890)

Correcting a typo as a way of driving off the spleen and regulating the circulation.

cliffordp committed:
correct 2nd param of useViewportMatch() Usage (#20911)

add missing opening single quote

voyager131 committed:
Update README.md (#20913)

There's an error in the code example. Please see discussion here for details:
https://wordpress.stackexchange.com/questions/360745/using-gutenberg-block-components-in-admin-interface-controls/360774#360774

Ringish committed:
Moves category multi select from LatestPosts to QueryControls (#20832)

* Implements FormTokenField

* Implements the posts query with multiple categories

* Improves code & resolves @draganescu's feedback

* Solves PHP formatting error

* Moves the category multiselect from LatestPosts to QueryControls

* Adds maxSuggestions prop and renames suggestions variable

* Resolves Lint error

* Moves logic from controlled component

etoledom committed:
Adding unit test to ListEdit to check for breacking changes from web (#20786)

koke committed:
Update Gutenberg

koke committed:
Update Gutenberg

koke committed:
Merge branch 'develop' into spt-design-fixes

draganescu committed:
testing larger margins (#20563)

koke committed:
Avoid page templates overwriting page title (#20865)

* Avoid templates overwriting page title

* Use logical OR to check for existing title

koke committed:
Update Gutenberg

koke committed:
Merge branch 'develop' into spt-no-overwrite-title

earnjam committed:
Fixes docblock for useViewportMatch (#20919)

koke committed:
Merge pull request #2015 from wordpress-mobile/spt-no-overwrite-title

Avoid page templates overwriting page title

jorgefilipecosta committed:
Fix: Cover block does not contain has-background-gradient class for predefined gradients

shiki committed:
Mobile: Add accessibility label to Block List Footer (#20633)

* Make Block List Footer inaccessible

* Add a11y label to Block List Footer

* Fix lint issue

* Change Block List Footer accessibility label

Adding “paragraph” would make it clearer and disambiguates it from the “Add block” button which behaves differently (shows a popup).

* block-list: Move import to correct section

hypest committed:
Update Gutenberg ref to use video-alt2 icon

gziolo committed:
Scripts: Update all webpack related dependencies (#20916)

hypest committed:
Add a release note for the take-a-video media option

adamziel committed:
Document default login credentials and `wp-env run` command (#20678)

* Document default login credentials and run command

* Add more descriptive CLI help messages

* Update README.md

* Update wording in `run` command description

* Fix typo in cli.js

* Fix linting errors

* Remove extra ;

* Whitespace

aduth committed:
Block Editor: Use useResizeObserver in place of direct react-resize-aware dependency (#20889)

aduth committed:
Update CODEOWNERS for aduth

vindl committed:
Site Editor: add Fullscreen mode (#20691)

Move FullscreenMode component to block-editor package so that it can
be reused both for Post Editor and Site Editor. In order to not have this tied
to same value in both cases, Site Editor store is added as part of this change
so that its fullscreen mode preference can be saved separately.

shiki committed:
Merge branch develop into issue/fix-footer-accessibility

shiki committed:
Merge pull request #1981 from wordpress-mobile/issue/fix-footer-accessibility

Add accessibility label to Block List Footer

jorgefilipecosta committed:
chore(release): publish

 - @wordpress/block-directory@1.5.5
 - @wordpress/block-editor@3.7.5
 - @wordpress/block-library@2.14.5
 - @wordpress/components@9.2.4
 - @wordpress/create-block@0.9.0
 - @wordpress/dependency-extraction-webpack-plugin@2.4.0
 - @wordpress/e2e-tests@1.12.7
 - @wordpress/edit-post@3.13.6
 - @wordpress/edit-site@1.3.4
 - @wordpress/edit-widgets@0.12.5
 - @wordpress/editor@9.12.5
 - @wordpress/format-library@1.14.5
 - @wordpress/list-reusable-blocks@1.13.4
 - @wordpress/nux@3.12.4
 - @wordpress/scripts@7.1.3
 - @wordpress/server-side-render@1.8.4

epiqueras committed:
FSE: Get current template part correctly for auto drafts (#20438)

Correctly gets the template part while auto-drafting template
parts. This fixes a PHP error that was echoed to the editor page.

pento committed:
Styles Panel: Don't force it to be closed by default. (#20617)

adamziel committed:
Fix/screen reader text (#20607)

* First stab at using VisuallyHidden component

* Completely replace screen-reader-text with <VisuallyHidden />

* Consistency changes with as="" property

* Remove extra space

* Formatting

* Formatting

* Restore .screen-reader-text CSS class in PHP files

* Use as="" instead of extra div

* Formatting change

* Use as="" instead of wrapping elements with screen-reader-text class

* Remove responsive-block-control/style.css from this PR

* Ensure <RichText tagName="figcaption" is always a direct descendant of <figure>

* Use label props when substituting for label with props

* Formatting

* Pass className to both VisuallyHidden and RichText

* Restore screen-reader-text in AdminNotices test

* Restore original formatting in admin-notices/test/index.js

youknowriad committed:
Refactor the inserter menu component and split into multiple smaller components (#20880)

Co-authored-by: Grzegorz (Greg) Ziółkowski <grzegorz@gziolo.pl>

koke committed:
Update page template picker after design review (#20883)

* Make title bar smaller

* Introduce new usePreferredColorSchemeStyle hook

* Add top space to template previews

* Adjust height of preview header and refactor a bit

* Fix typo in useViewportMatch documentation

Co-authored-by: Pinar Olguc <pinarolguc@gmail.com>

koke committed:
Update Gutenberg

koke committed:
Merge branch 'develop' into spt-design-fixes

koke committed:
Merge pull request #2020 from wordpress-mobile/spt-design-fixes

Update page template picker after design review

ceyhun committed:
Merge pull request #1976 from wordpress-mobile/try/speed-up-uitests

Speed up e2e tests execution

aladin002dz committed:
Remove inaccurate message from image block (#20909)

hypest committed:
Update Gutenberg ref

youknowriad committed:
Fix block preview

hypest committed:
Update Gutenberg ref

Commit:
Update Gutenberg ref

aduth committed:
Compat: Conditionally filter editor settings for image dimensions (#20939)

* Compat: Conditionally filter editor settings for dimensions

* Compat: Use `empty` to detect imageSizes

`is_array` still warns if not set. Coincidentally helps avoid an unnecessary call to wp_get_registered_image_subsizes when there's no sizes to work with. One difference is that imageDimensions would not be set unless imageSizes is non-empty. But if the idea is for imageDimensions to be dependent upon imageSizes here, it's a sensible caveat.

* Framework: Convert comment to multiline

Resolves lint errors

Commit:
RNMobile - Disable Cover for production (#20965)

Commit:
Update Gutenberg ref

talldan committed:
Remove iframe from embedded content elements (#20976)

talldan committed:
Escape class name to avoid malformed html tag (#20977)

garretthyder committed:
Lowercase visual editor and code editor to match block editor and classic editor. (#20968)

spacedmonkey committed:
Add menus endpoints. (#20292)

* Add menus endpoints.

* Fix lints.

* Fix unit tests.

* Fix unit tests again.

* Final unit test fix.

Commit:
Merge pull request #2028 from wordpress-mobile/fix/cover-block-dev-only

Disable Cover Block for Production

Commit:
RNMobile - FloatingToolbar - Make children within the toolbar accessible with VoiceOver / TalkBack (#20789)

Commit:
Merge branch 'develop' of github.com:wordpress-mobile/gutenberg-mobile into fix/floating-toolbar-accessibility

# Conflicts:
#	gutenberg

Commit:
Update Gutenberg ref

epiqueras committed:
Block Library: Add a Post Comments block. (#19581)

Co-authored-by: Noah Allen <noahtallen@gmail.com>

Commit:
Merge pull request #1997 from wordpress-mobile/fix/floating-toolbar-accessibility

FloatingToolbar - Make children accessible

gziolo committed:
Add visible labels to BlockPatternPicker pattern selection buttons (#19789)

* Add visible labels to InnerBlocksTemplatePicker pattern selection buttons
Fixes #18737.

* Adjust UI for Block Variation Picker labels

Co-authored-by: Jon Q <hello@jonquach.com>

johnstonphilip committed:
Load custom blocks in the Site Editor page (#20549)

jasmussen committed:
Polish a few icons (#20980)

This updates icons in the component with recent refinements from the Figma file.

- bold
- italic
- strikethrough
- indent
- outdent
- spacer

jasmussen committed:
Fix mobile header. (#20946)

ellatrix committed:
Fix input rules (#20964)

* Fix input rules

* Add e2e tests

earnjam committed:
Try: Improve permalink editing (#12009)

Allows immediate permalink editing after initializing the editor.

* Filter REST API to return permalink_template

* Add new getEditedPostSlug selector

* Update auto-generated docs

* Make permalink editing more timely/accurate in title editor

* Make permalink editing more timely/accurate in sidebar panel

* Update e2e tests to reflect that panel should always show

* Add unit tests for getEditedPostSlug selector

hypest committed:
Update Gutenberg ref

jbinda committed:
apply flex to prevent caption overlap in Gallery block (#20987)

hypest committed:
Update Gutenberg ref

hypest committed:
Merge branch 'develop' into issue/1697-improve-take-video-icon

youknowriad committed:
Add template preview to the edit site template switcher (#20958)

jbinda committed:
Fix: Fix Gallery caption overlapped by other block (#2033)

* update ref to master (fix gallery caption)

nerrad committed:
Convert _experimentalCreateInterpolateElement to a stable api. (#20699)

* make createInterpolateElement stable

* implement new stable function everywhere it was implemented

* update changelog

hypest committed:
[RNmobile] Improve "take video" icon (#20927)

* Port video-alt2 icon from dashicons

* Use video-alt2 icon on the media selection sheet

* Better name and design for the capture video icon

* Update unit test snapshot

* Update the capture-a-photo icon on mobile media option picker

* Use the 'image' icon for the choose-from-device option

* Some code formatting adjustment

* Update unit test snapshot

* Update vectors.

* Update snapshot.

Co-authored-by: jasmussen <joen@automattic.com>

hypest committed:
Point to the merged Gutenberg PR

hypest committed:
Merge branch 'develop' into issue/1697-improve-take-video-icon

hypest committed:
Merge pull request #2024 from wordpress-mobile/issue/1697-improve-take-video-icon

Update Gutenberg ref to use video-alt2 icon

jasmussen committed:
Try: Nicer block footprint for social links. (#20978)

* Try: Nicer block footprint for social links.

* Address feedback

SergioEstevao committed:
Rnmobile/fix space max2 (#20986)

* Disable Cover for production

* Adapt max according to the current value.

Co-authored-by: Marko Savic <savicmarko1985@gmail.com>

talldan committed:
Make the inner button block not allowed as a reusable block or… (#20948)

* Make the inner button block not usuable as a reusable block

* Also disallow HTML editing

* Remove restriction on editing button as html

draganescu committed:
[ Latest Posts ] Fix link for read more markup (#20917)

mapk committed:
Update glossary (#20934)

* Update glossary

Added "Block patterns" and "Template parts". Also updated the Block Inserter to Block Library.

* Fixed spacing issue after period.

Soean committed:
Fix link control search results spacing. (#21003)

* Fix search results spacing.

* Remove base control spacing

Commit:
RangeControl: Improve disabled rendering and interactions (#20723)

This update improves the UI rendering of the RangeControl, sepcifically by muting the Rail, Track, and Marks. The disabled prop is also passed along to the inner (number) Input and Reset value button.

scruffian committed:
Block Styles: Remove the block margin in the style selector (#19983)

pento committed:
Blocks: Allow the Default Style selector to be hidden. (#20620)

Co-authored-by: Marcus Kazmierczak <marcus@mkaz.com>

aduth committed:
Project Management Automation: Add TypeScript type-checking (#20850)

* Project Management Automation: Add TypeScript type-checking

* Project Management Automation: Remove unused typedefs

marecar3 committed:
[RNMobile] - Implements Dark Mode on Android platform (#19293)

Copons committed:
Make sure items with long names can fit in the Autocomplete component (#20962)

Soean committed:
Block patterns: improve success notice (#21005)

* Add period at the end of sentence.

* Add translator comment

marecar3 committed:
Feature/1675 dark mode android integration (#1717)

ellatrix committed:
Paste: replace iframes with url (#20983)

ellatrix committed:
Editable Component (#18361)

* Text Component

* Rename

* Lint fixes

SergioEstevao committed:
Add isFileURL method and use it on all native media upload checks. (#20985)

* Add isFileURL method and use it on all native media upload checks.

* Check if url is defined.

* Add test for undefined and fix lint error for isFileURL

* Replace isFileURL method for getProcotol == 'file' mettod

talldan committed:
Add basic nav block example for inserter and styles previews (#21011)

Tug commented:
@dratwas thanks for reviewing! I might actually update once more since we're at it (also gutenberg-mobile changes are out of sync with gutenberg here and it causes problem when we try to run the app)

@gziolo Just wanted you to know about it :)
It's a lot but it shouldn't be modifying any web code, most of the diff is simply bringing the monorepo branch up to date with `master` (we can't easily rebase given the amount of commits this branch has and the never ending rebase hell it would cause)

Tug committed:
Merge remote-tracking branch 'origin/master' into rnmobile/monorepo-update-gb-mobile-rn

Tug committed:
Update manifest doc

Tug commented:
@gziolo Actually we'll need an admin to merge without squashing. Let's ping once it's ready

Tug committed:
Merge commit 'e288c45e882db9808e0396715f379edb07f32416' into rnmobile/monorepo-update-gb-mobile-rn

Tug committed:
Merge remote-tracking branch 'origin/master' into rnmobile/monorepo-update-gb-mobile-rn

Tug committed:
Prevent import loop in _colors.scss

Tug committed:
Fix web build and convert Aztec from flow to js

Tug committed:
Fix npm run test-unit

Tug committed:
Add back src folder to autom import in sass transformer

Tug commented:
@dratwas it's ready for another pass :)

Btw, you can try to actually build the dev bundle by requesting:
http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false
http://localhost:8081/index.bundle?platform=android&dev=true&minify=false

ceyhun reviewed (approved):
I was able to successfully run all the tests 👍

gziolo commented:
> @gziolo Actually we'll need an admin to merge without squashing. Let's ping once it's ready

It's disabled for me at the moment. @youknowriad has superpowers to change that :)

Tug commented:
@hypest I think you have the superpowers as well, could you try to do it?

For reference, the reason we need this is because until we haven’t made the switch to use `gutenberg` as our main repo, we’ll want to pull changes from `gutenberg-mobile` with `git subtree pull` and that won’t work if we rewrite history!

hypest commented:
Merging with "Merge commit" option set to true 👍
